app/javascript/App/bundles/CoinShow/index.js

```diff
@@ -1,369 +1,979 @@
-        marginBottom: `${theme.spacing.unit * 1.5}px`,
-      },
-    },
-    coinName: {
-      fontSize: '1.5rem',
-      fontWeight: 'bold',
-      [theme.breakpoints.up('md')]: {
-        marginRight: `${theme.spacing.unit}px`,
-      },
-      [theme.breakpoints.down('sm')]: {
-        flex: '1 0 50%',
-        paddingRight: `${theme.spacing.unit * 0.75}px`,
-        textAlign: 'right',
-        marginBottom: `${theme.spacing.unit * 0.5}px`,
-      },
-    },
-    coinSymbol: {
-      fontSize: '1rem',
-      [theme.breakpoints.up('md')]: {
-        marginRight: `${theme.spacing.unit * 1.5}px`,
-      },
-      [theme.breakpoints.down('sm')]: {
-        flex: '1 0 50%',
-        paddingLeft: `${theme.spacing.unit * 0.75}px`,
-        textAlign: 'left',
-        marginBottom: `${theme.spacing.unit * 0.5}px`,
-      },
-    },
-    coinPrice: {
-      fontSize: '1rem',
-      fontWeight: 'bold',
-      [theme.breakpoints.up('md')]: {
-        marginRight: `${theme.spacing.unit * 1.5}px`,
-      },
-      [theme.breakpoints.down('sm')]: {
-        flex: '1 0 33%',
-        textAlign: 'center',
-      },
-    },
-    coinChange: {
-      fontSize: '0.8rem',
-      [theme.breakpoints.up('md')]: {
-        marginRight: `${theme.spacing.unit * 1.5}px`,
-      },
-      [theme.breakpoints.down('sm')]: {
-        flex: '1 0 33%',
-        textAlign: 'center',
-      },
-    },
-    watchButtonContainer: {
-      [theme.breakpoints.down('sm')]: {
-        flex: '1 0 33%',
-        textAlign: 'center',
-      },
-    },
-    watchedButton: {
-      backgroundColor: '#40a9ff',
-      color: '#fff',
-    },
-    unwatchedButton: {
-      backgroundColor: '#fff',
-      color: '#40a9ff',
-    },
-    mainCard: {
-      padding: 0,
-      marginBottom: `${theme.spacing.unit * 2}px`,
-      '&:last-child': {
-        marginBottom: 0,
-      },
-    },
-    subCard: {
-      padding: 0,
-      marginBottom: `${theme.spacing.unit * 2}px`,
-      '&:last-child': {
-        marginBottom: 0,
-      },
-    },
-    subCardHeader: {
-      paddingBottom: 0,
-    },
-  })
+const NewsLabel = withStyles(styles)(({ classes }) => {
+  return (
+    <React.Fragment>
+      <span>News</span>
+      <Icon
+        name="external-link-alt"
+        regular
+        className={classes.newsLabelIcon}
+      />
+    </React.Fragment>
+  )
+})
 
 class CoinShow extends Component {
   chart = undefined
 
-  state = {
-    liveCoinArr: [],
-    currency: 'USD',
-    watched: this.props.watching,
-    iconLoading: false,
-  }
+  constructor(props) {
+    super(props)
 
-  watchlistHandler(coin) {
-    if (window.location === `/coins/${coin.get('name')}`) {
-      window.location = `/coins/${coin
-        .get('name')
-        .replace(/ /, '-')
-        .toLowerCase()}`
+    const { tokenMetrics } = props
+    const hasTokenMetrics = !_.isEmpty(tokenMetrics)
+    const hashTag = _.get(props, ['location', 'hash'], '').slice(1) // remove prepended octothorpe
```

Comment:
I'm very happy that our codebase has the word "octothorpe" in it now :octocat: 🐙